plugins/org.eclipse.fordiac.ide.model/src/org/eclipse/fordiac/ide/model/libraryElement/impl/FBNetworkAnnotations.java

```diff
@@ -181,13 +188,24 @@ private static Diagnostic createInterfaceBarCollisionDiagnostic(final FBNetworkE
 						LibraryElementPackage.Literals.POSITIONABLE_ELEMENT__POSITION));
 	}
 
-	private static int getIntPreference(final Map<Object, Object> context, final String name) {
-		return ((Integer) context.computeIfAbsent(name, FBNetworkAnnotations::internalGetIntPreference)).intValue();
+	private static IProject getRootProject(final EObject eObject) {
+		if (eObject == null) {
+			return null;
+		}
+		final EObject root = EcoreUtil.getRootContainer(eObject);
+		if (root != null && root instanceof final LibraryElement libElem && libElem.getTypeLibrary() != null) {
```

Comment:
Both `EcoreUtil.getRootContainer` and `instanceof` are null-safe, so we can safely simplify to:
```suggestion
		if (EcoreUtil.getRootContainer(eObject) instanceof final LibraryElement libElem && libElem.getTypeLibrary() != null) {
```